src/uct/ib/base/ib_md.c

```diff
@@ -285,13 +298,125 @@ static void uct_ib_md_print_mem_reg_err_msg(ucs_log_level_t level, void *address
     ucs_log(level, "%s", msg);
 }
 
+void *uct_ib_md_thread(void *arg) {
+    uct_ib_md_thread_ctx_t *ctx = arg;
+    size_t size;
+    int i = 0;
+    int ret;
+
+    while (ctx->len) {
+        size = ucs_min(ctx->len, UCT_IB_MD_MAX_MR_SIZE);
+        if (ctx->access != UCT_IB_MEM_DEREG) {
+            ctx->mr[i] = ibv_reg_mr(ctx->pd, ctx->addr, size, ctx->access);
```

Comment:
UCS_PROFILE
also if we use the basic ibv_reg_mr, why access is uint64_t and not int?